src/main/java/com/google/sps/servlets/MemesServlet.java

```diff
@@ -81,11 +81,11 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Storage storage = StorageOptions.newBuilder().setProjectId(PROJECT_ID).build().getService();
     Bucket bucket = storage.get(BUCKET_NAME);
     Page<Blob> blobs = bucket.list();
-
+    String imgTag = "";
     // Output <img> elements as HTML.
     for (Blob blob : blobs.iterateAll()) {
-      String imgTag = String.format("<img src=\"%s\" width = \"300\"/>", blob.getMediaLink());
-      response.getWriter().println(imgTag);
+      imgTag = imgTag + String.format("<a href=\"" + "https://spring21-sps-43.appspot.com/share?meme=" + blob.getName() + "\"><img src=\"%s\" width = \"300\"/></a>", blob.getMediaLink());
```

Comment:
Whenever we redirect the user within our own code, use relative links, like "/share?meme=", instead of hard coded links, like "https://spring21-sps-43.appspot.com/share?meme=". This will make sure the server points to the right address, no matter if we're using a local server or live server, or if we ever change our server's url.